### DISCUSSION TITLE
Submit cart, address and payment details in a single POST request

### DISCUSSION THREAD

**Discussion by timeverts** [craftcms/commerce#2330] [Ideas]
### Description
This is a feature request. 

Please make it possible to submit a cart (line items), shipping/billing details and payment details all in a single POST request (e.g. for a true single page shopping experience). 

Currently it's not possible to do so because `PaymentsController::actionPay()` is the method to call for making payment and it doesn't seem to support all that info being posted at once. The best it can do is look for an existing cart/order in the session which means the cart would have to have been submitted first via a separate request.

I realise the single page shopping experience can be achieved using a couple of Ajax requests prior to submitting the payment request, but it would be great if it could be achieved without Ajax too.

**Comment by lukeholder** [comment]
Thanks for the feature request. Hope to see others 👍  this if it is important to them.
